### PR TITLE
Bug 1986840: track number of active alertmanager receivers via telemetry

### PR DESCRIPTION
**Pull request by paulfantom**
As in title.

Data is approved for telemetry as per conversation - https://coreos.slack.com/archives/C0VMT03S5/p1623319743365200. relevant msg - https://coreos.slack.com/archives/C0VMT03S5/p1623336558389100?thread_ts=1623319743.365200&cid=C0VMT03S5

* [ ] I added CHANGELOG entry for this change.
* [ ] No user facing changes, so no entry in CHANGELOG was needed.

putting on hold as this targets 4.9

cc @simonpasquier 

/hold

### CODE REVIEW

File: jsonnet/rules.libsonnet [hunk 1]
```diff
@@ -320,6 +320,10 @@ local droppedKsmLabels = 'endpoint, instance, job, pod, service';
             record: 'cluster:control_plane:all_nodes_ready',
             // Returns 1 if all control plane nodes are ready and is absent otherwise. Should be used to suppress alerts during control plane upgrades or disruption.
           },
+          {
+            expr: 'count(alertmanager_receivers)',
```
Comment (simonpasquier) [inline jsonnet/rules.libsonnet:324]: I'd rather look at the max value in the openshift-monitoring namespace.

```suggestion
            expr: 'max(alertmanager_receivers{namespace="openshift-monitoring"})',
```

**Review by simonpasquier — COMMENTED** [review]
can we also get rid of the `cluster:alertmanager_routing_enabled:max` recording rule and modify `AlertmanagerReceiversNotConfigured` to use the new metric instead?

File: jsonnet/rules.libsonnet (hunk 1 above)
Comment (paulfantom) [reply jsonnet/rules.libsonnet:324]: SGTM, fixed.

**Comment by paulfantom** [comment]
Done :)

**Review by simonpasquier — COMMENTED** [review]
/lgtm

**Comment by simonpasquier** [comment]
/hold
The `AlertmanagerReceiversNotConfigured` alert doesn't fire anymore, we need to use `alertmanager_integrations` rather than `alertmanager_receivers`.

**Comment by paulfantom** [comment]
Refactored to use `alertmanager_integrations`.

/unhold

File: Documentation/data-collection.md [hunk 2]
```diff
@@ -253,6 +253,10 @@ data:
     # We want to have an idea of how loaded our api server(s) are globally.
     - '{__name__="cluster:apiserver_current_inflight_requests:sum:max_over_time:2m"}'
     #
+    # (monitoring-team) cluster:alertmanager_integrations:max tracks the total number of active alertmanager receivers
```
Comment (simonpasquier) [inline Documentation/data-collection.md:256]:
```suggestion
    # (monitoring-team) cluster:alertmanager_integrations:max tracks the total number of active alertmanager integrations
```

**Pull request merged**